runtime-modules/storage/src/tests/mod.rs

```diff
@@ -6242,3 +6242,98 @@ fn get_upload_costs_succeeds_overflow_sat() {
         assert_eq!(funds_needed_saturated, u64::MAX);
     });
 }
+
+#[test]
+fn uploading_objects_with_invalid_cid_length_should_fail() {
+    build_test_externalities().execute_with(|| {
+                let starting_block = 1;
+        run_to_block(starting_block);
+
+        let initial_balance = 1000;
```

Comment:
Should be `ed() + data_object_state_bloat_bond + size_fee` (where size fee is calculated based on data object size, or `0` if `Storage::data_object_per_mega_byte_fee()` is also `0`)